src/VisualStudio/VisualStudioDiagnosticsToolWindow/VisualStudioDiagnosticsWindowPackage.cs

```diff
@@ -75,17 +77,19 @@ private void ShowToolWindow(object sender, EventArgs e)
         /// Initialization of the package; this method is called right after the package is sited, so this is the place
         /// where you can put all the initialization code that rely on services provided by VisualStudio.
         /// </summary>
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
+            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            var componentModel = (IComponentModel)GetService(typeof(SComponentModel));
+            var componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(true);
```

Comment:
so, how the ordering works? let's say multiple packages (A, B, C) are getting loaded and InitializeAsync are called, and they try to move to UI threads in order, A, B, C. does system make sure that A, B, C gets UI threads in order, or it can get UI threads in arbitary order like, C, B, A even though packages started loading A, B, C?

asking since worrying that something like componentModel.GetService<VisualStudioWorkspace>() somehow run first than initializing ForegroundObject or scheduler from Roslyn packages initializeAsync even though Roslyn package is loaded first.

Reply:
Honestly I have no idea. I know how `ForegroundThreadAffinitizedObject` is initialized in test code but not in production.